src/SprykerEco/Zed/Heidelpay/Business/Payment/PaymentMethodFilter.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+/**
+ * MIT License
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace SprykerEco\Zed\Heidelpay\Business\Payment;
+
+use ArrayObject;
+use Generated\Shared\Transfer\PaymentMethodsTransfer;
+use Generated\Shared\Transfer\PaymentMethodTransfer;
+use Generated\Shared\Transfer\QuoteTransfer;
+use SprykerEco\Zed\Heidelpay\Dependency\Facade\HeidelpayToMoneyFacadeInterface;
+use SprykerEco\Zed\Heidelpay\HeidelpayConfig;
+
+class PaymentMethodFilter implements PaymentMethodFilterInterface
+{
+    protected const HEIDELPAY_EASY_CREDIT_PAYMENT_METHOD = 'heidelpayEasyCredit';
+
+    /**
+     * @var \SprykerEco\Zed\Heidelpay\HeidelpayConfig
+     */
+    protected $config;
+
+    /**
+     * @var \SprykerEco\Zed\Heidelpay\Dependency\Facade\HeidelpayToMoneyFacadeInterface
+     */
+    protected $moneyFacade;
+
+    /**
+     * @param \SprykerEco\Zed\Heidelpay\HeidelpayConfig $config
+     * @param \SprykerEco\Zed\Heidelpay\Dependency\Facade\HeidelpayToMoneyFacadeInterface $moneyFacade
+     */
+    public function __construct(
+        HeidelpayConfig $config,
+        HeidelpayToMoneyFacadeInterface $moneyFacade
+    ) {
+        $this->config = $config;
+        $this->moneyFacade = $moneyFacade;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\PaymentMethodsTransfer $paymentMethodsTransfer
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     *
+     * @return \Generated\Shared\Transfer\PaymentMethodsTransfer
+     */
+    public function filterPaymentMethods(
+        PaymentMethodsTransfer $paymentMethodsTransfer,
+        QuoteTransfer $quoteTransfer
+    ): PaymentMethodsTransfer {
+
+        $result = new ArrayObject();
+        $grandTotal = $this->moneyFacade->convertIntegerToDecimal($quoteTransfer->getTotals()->getGrandTotal());
+        foreach ($paymentMethodsTransfer->getMethods() as $paymentMethod) {
+            $address = $this->isAddressCorrect($quoteTransfer);
+            if ($this->isPaymentMethodHeidelpayEasyCredit($paymentMethod) && $this->isTotalOutOfRange($grandTotal) && $this->isAddressCorrect($quoteTransfer)) {
```

Comment:
It would be great if you explain me, what we actually filter here

Reply:
See [ECO-1539] "When will this payment option be shown".